File: src/main/java/com/example/demoapplication/service/UserService.java
package com.example.demoapplication.service;

import com.example.demoapplication.dto.UserDTO;
import com.example.demoapplication.entity.User;
import com.example.demoapplication.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService {


    @Autowired
    private UserRepository userRepository;

    //CREATE -> Accept DTO
    public User createUser(UserDTO userDTO)
    {
        User user = new User();
        user.setName(userDTO.getName());
        user.setEmail(userDTO.getEmail());
        user.setAge(userDTO.getAge());

        return userRepository.save(user);
    }

    //FindAll
    public List<User> getAllUsers()
    {
        return userRepository.findAll();
    }


    //GetById
    public User getUserById(Long id)
    {
       return  userRepository.findById(id).orElseThrow(()-> new RuntimeException("User not found"));
    }


    //UpdateUserbYId
    public User updateUser(Long id,UserDTO userDTO)
    {
        User existingUser = userRepository.findById(id).orElseThrow(()->new RuntimeException("user not found"));

        existingUser.setName(userDTO.getName());
        existingUser.setEmail(userDTO.getEmail());
        existingUser.setAge(userDTO.getAge());

       return userRepository.save(existingUser);

    }

    //DeleteUser
    public void deleteUser(Long id)
    {
        userRepository.deleteById(id);
    }
}

File: frontend/src/services/userService.js
import axios from "axios";

const BASE_URL = "http://localhost:8080/Demo";

export const addUser = (user) => axios.post(`${BASE_URL}/add`, user);

export const getAllUsers = () => axios.get(`${BASE_URL}/findAll`);

export const getUserById = (id) => axios.get(`${BASE_URL}/find/${id}`);

export const updateUser = (id, user) =>
  axios.put(`${BASE_URL}/update/${id}`, user);

export const deleteUser = (id) =>
  axios.delete(`${BASE_URL}/delete/${id}`);
